control-plane/stor-port/src/types/v0/transport/policies.rs

```diff
@@ -0,0 +1,86 @@
+use crate::types::v0::transport::Nexus;
+
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+
+/*
+    This file defines the policies that control plane uses to make
+    decisions about rebuild workflow behaviour in case of a child
+    becoming faulted. We can logically provide multiple diferent policies.
+    These policies are represented as an enum, since only one of the policy
+    can be attached to a given volume/nexus at a time, and this attachment
+    will be done during provisioning. Once attached, the further behaviour
+    of the policy can vary depending on runtime state of the nexus.
+    Any new policy introduced in future will have to implement Policy trait
+    for defining the policy behaviour.
+*/
+
+// time constants, in seconds
+const TWAIT_SPERF: std::time::Duration = Duration::new(600, 0);
+const TWAIT_SAVAIL: std::time::Duration = Duration::new(300, 0);
+const TWAIT_ZERO: std::time::Duration = Duration::new(0, 0);
+// 100GiB = 100 * 1024 * 1024 * 1024 bytes
+const VOL_SIZE_100GIB: u64 = 107374182400;
+
+#[derive(Serialize, Deserialize, Debug, Clone, Copy, Eq, PartialEq)]
+// SystemPerf policy
+pub struct SystemPerf {}
+#[derive(Serialize, Deserialize, Debug, Clone, Copy, Eq, PartialEq)]
+// SystemAvail policy
+pub struct SystemAvail {}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone, Copy)]
+pub enum RuleSet {
+    SystemPerf(SystemPerf),
+    SystemAvail(SystemAvail),
+}
+
+impl RuleSet {
+    // Returns a duration value. The caller can use this duration
+    // to timeout between two time Instants, rather than this interface
+    // returning the absolute Instant to compare with.
+    pub fn faulted_child_wait(&self, nexus: &Nexus) -> Duration {
+        match self {
+            RuleSet::SystemAvail(pol) => pol.faulted_child_wait_duration(nexus),
+            RuleSet::SystemPerf(pol) => pol.faulted_child_wait_duration(nexus),
+        }
+    }
+
+    pub fn assign(size: u64) -> Self {
+        match size > VOL_SIZE_100GIB {
```

Comment:
I thought we decided on the hl meeting not to do policies based on anything yet, has this been revised?

Reply:
Not revised, I just sanity tested it for both cases this way. For now I'll modify assign() call to take only default() path.